addons/account/models/account_move_line.py

```diff
@@ -2903,6 +2924,24 @@ def remove_move_reconcile(self):
         """ Undo a reconciliation """
         (self.matched_debit_ids + self.matched_credit_ids).unlink()
 
+    def _reconcile_marked(self):
+        temp_numbers = list({
+            line.matching_number
+            for line in self
+            if line.matching_number and line.matching_number.startswith('I')
+        })
+        if temp_numbers:
+            for _matching_number, account, lines in self._read_group(
```

Comment:
I suppose it's faster to compute temp_numbers first, as the fields are already fetched, rather than skipping the above code and directly use the domain `[('matching_number', 'ilike', 'I%')]`  in the _read_group?  If that's the reason, you should add a comment to make it explicit :)

Reply:
It is faster, but we also want to only consider the moves being posted to avoid checking and fetching everything each time